clippy_lints/src/macro_metavars_in_unsafe.rs

```diff
@@ -220,11 +220,10 @@ impl<'tcx> LateLintPass<'tcx> for ExprMetavarsInUnsafe {
         // `check_stmt_post` on `(Late)LintPass`, which we'd need to detect when we're leaving a macro span
 
         let mut vis = BodyVisitor {
-            #[expect(clippy::bool_to_int_with_if)] // obfuscates the meaning
-            expn_depth: if body.value.span.from_expansion() { 1 } else { 0 },
             macro_unsafe_blocks: Vec::new(),
+            expn_depth: if body.value.span.from_expansion() { 1 } else { 0 },
```

Comment:
The `#[expect]` should have moved along with the field. That explains why you had to add your third commit, as you lost the attribute.

You should at the minimum re-add the attribute in the second commit and drop the third commit. But the attribute situation should be examined more closely.

Reply:
Thanks, @samueltardieu.

Reply:
This problem should be resolved.

Reply:
Do you mean manually, or automatically? Will attributes now move with the fields, or is this a bug that needs fixing? (I don't seem to find tests with this case that we now know is problematic)